### PR TITLE
Bug 1169309 - Separate the concat from the pretranslate step

### PR DESCRIPTION
**Pull request by stasm**
https://bugzilla.mozilla.org/show_bug.cgi?id=1169309

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=4135206124dd709772deca01bd4e0e9293fa28c6

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8931852589407291da9e523247097f49281f71a3

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=233b93c952ed9c39484ac88c4e7f107a5dfa5ccf

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f24fc123f3fe5f23d05e49b0154e91f65a856b47

File: build/webapp-optimize.js [hunk 1]
```diff
@@ -40,72 +40,76 @@ var HTMLOptimizer = function(options) {
...
-       !this.win.document.querySelector('link[rel="localization"]')) ||
-      ignore[this.webapp.sourceDirectoryName]) {
-    return;
-  }
-
-  // Since l10n.js was read before the document was created, we need to
-  // explicitly initialize it again via mozL10n.bootstrap, which looks for
-  // rel="localization" links in the HTML and sets up the localization context.
-  mozL10n.bootstrap(this.webapp.url, this.config);
-  this._optimize();
```
Comment (stasm) [inline build/webapp-optimize.js]: @rickychien I realized that `this._optimize` was only called when the document passed the `if` test above, i.e. it was localization-enabled.  Is this intended?  I'll keep this the same way in the PR, but I'm wondering if it's a bug.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1c67ed031a82827027c3dcfc581435b272fc53e3

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8b81b605dd9999af015c3852665d7230f9652bea

**Pull request merged**